packages/consumption/src/modules/requests/itemProcessors/readAttribute/ReadAttributeRequestItemProcessor.ts

```diff
@@ -36,14 +38,34 @@ export class ReadAttributeRequestItemProcessor extends GenericRequestItemProcess
         if (parsedParams.isWithExistingAttribute()) {
             const foundAttribute = await this.consumptionController.attributes.getLocalAttribute(parsedParams.existingAttributeId);
 
-            if (!foundAttribute) {
+            if (typeof foundAttribute === "undefined") {
                 return ValidationResult.error(TransportCoreErrors.general.recordNotFound(LocalAttribute, requestInfo.id.toString()));
             }
 
             const ownerIsCurrentIdentity = this.accountController.identity.isMe(foundAttribute.content.owner);
             if (!ownerIsCurrentIdentity && foundAttribute.content instanceof IdentityAttribute) {
                 return ValidationResult.error(CoreErrors.requests.invalidRequestItem("The given Attribute belongs to someone else. You can only share own Attributes."));
             }
+
+            const latestSharedVersion = await this.consumptionController.attributes.getSharedVersionsOfRepositoryAttribute(
+                parsedParams.existingAttributeId,
+                [requestInfo.peer],
+                true
+            );
+            if (latestSharedVersion.length > 0) {
+                if (typeof latestSharedVersion[0].shareInfo?.sourceAttribute === "undefined") {
+                    throw new Error(`The Attribute ${latestSharedVersion[0].id} does not fulfill the requirements of an own shared Attribute.`);
+                }
+
+                const latestSharedVersionSourceAttribute = await this.consumptionController.attributes.getLocalAttribute(latestSharedVersion[0].shareInfo.sourceAttribute);
+                if (typeof latestSharedVersionSourceAttribute === "undefined") {
+                    throw new Error(`The Attribute ${latestSharedVersion[0].shareInfo.sourceAttribute} was not found.`);
+                }
+
+                if (await this.consumptionController.attributes.isASuccessorOf(latestSharedVersionSourceAttribute, foundAttribute)) {
+                    return ValidationResult.error(CoreErrors.requests.invalidRequestItem("You cannot share the predecessor of an already shared Attribute version."));
+                }
```

Comment:
Is there a reason you sometimes throw an error while at other times you return a `ValidationResult`?

Reply:
I return a `ValidationResult`, if the user made a mistake (by entering a response that is not allowed), which we might expect and can handle. However, if there is something broken in the data which we do not expect to happen, I throw an error (so the user doesn't assume they made a mistake).